src/lsqecc/pauli_rotations/circuit.py

```diff
@@ -356,7 +356,12 @@ def load_reversible_from_qasm_string(quasm_string: str) -> "PauliOpCircuit":
 
     @staticmethod
     def join(lhs: "PauliOpCircuit", rhs: "PauliOpCircuit") -> "PauliOpCircuit":
-        assert lhs.qubit_num == rhs.qubit_num
+        try:
+            assert lhs.qubit_num == rhs.qubit_num
+        except AssertionError:
+            print("The two circuits joined must have the same number of qubits!")
+            # Dummy return for making tests
+            return lhs
```

Comment:
I think we should not omit the error here. You can have a test case that trigger the error and use `with pytest.raises(your_error_type):`

Reply:
I am thinking:
```suggestion
        if lhs.qubit_num != rhs.qubit_num:
            raise Exception("The two circuits joined must have the same number of qubits!")
```
And the test case would be:
```
def test_join_different_qubit_num:
    # some code to initialize the test
    with pytest.raises(Exception):
        # Calling join() on conditions that will raise the exception 
```